BeeTracker/app/src/main/java/com/blueprint/foe/beetracker/MainActivity.java

```diff
@@ -26,95 +28,159 @@
-
-            @Override
-            public void onError(FacebookException exception) {
-                Log.e(TAG, "There was an error on Facebook login. " + exception.toString());
-                exception.printStackTrace();
-                showErrorDialog(getString(R.string.error_message_login));
-            }
         });
 
-
-        loginCallback = new Callback<BeeTrackerCaller.SignupResponse>() {
+        emailPasswordLoginCallback = new Callback<BeeTrackerCaller.EmailPasswordSigninResponse>() {
             @Override
-            public void onResponse(Call<BeeTrackerCaller.SignupResponse> call, Response<BeeTrackerCaller.SignupResponse> response) {
-                if (response.code() == 401 || response.body() == null ||  response.body().getToken() == null) {
+            public void onResponse(Call<BeeTrackerCaller.EmailPasswordSigninResponse> call, Response<BeeTrackerCaller.EmailPasswordSigninResponse> response) {
+                if (response.code() == 401 || response.headers() == null) {
                     Log.e(TAG, "The response from the server is 401 + " + response.message());
                     LoginManager.getInstance().logOut();
                     spinningIconDialog.dismiss();
                     showErrorDialog(getString(R.string.error_message_login));
+                    setInputFieldsActiveStateToError(emailInput, passwordInput);
                     return;
                 }
                 SharedPreferences sharedPref = getSharedPreferences(getString(R.string.preference_file_key), Context.MODE_PRIVATE);
-                sharedPref.edit().putString(getString(R.string.preference_login_token), response.body().getToken()).commit();
+                sharedPref.edit().putString(getString(R.string.preference_login_token), response.headers().get("access-token")).commit();
+                sharedPref.edit().putString(getString(R.string.preference_login_token_type), response.headers().get("token-type")).commit();
+                sharedPref.edit().putString(getString(R.string.preference_login_client), response.headers().get("client")).commit();
+                sharedPref.edit().putString(getString(R.string.preference_login_expiry), response.headers().get("expiry")).commit();
+                sharedPref.edit().putString(getString(R.string.preference_login_uid), response.headers().get("uid")).commit();
                 navigateToHome();
             }
 
             @Override
-            public void onFailure(Call<BeeTrackerCaller.SignupResponse> call, Throwable t) {
-                Log.e(TAG, "There was an error with the loginCallback + " + t.toString());
+            public void onFailure(Call<BeeTrackerCaller.EmailPasswordSigninResponse> call, Throwable t) {
+                Log.e(TAG, "There was an error with the login callback + " + t.toString());
                 t.printStackTrace();
                 LoginManager.getInstance().logOut();
                 spinningIconDialog.dismiss();
```

Comment:
Please check if spinningIconDialog is null (sorry these aren't your changes, my bad again :/ )